src/common.c

```diff
@@ -59,6 +59,8 @@ void common_init(void)
 int ishex(const char *q)
 {
 	const char *p=q;
+	if (!q || !*q)
+		return 0;
```

Comment:
I assume these fixes are unrelated to the PR (although perhaps found with it)? They should be a separate commit and PR.

Reply:
Ok, problems found using this PR should not be placed into it?  That means I will have to make a babied fix PR for the TS, then pre-fix things, then improve the TS based upon those fixes???

I can do things that way, for sure, if that is how you believe it should be handled.

I did have questions on the ishex() type stuff    We may have had discussions about this at a prior time, I simply do not remember.    As the code was written, things like ishex("") returned true.  I would argue this is wrong, UNLESS, there was a specific function that was used where null was OK (just like we have special functions which allow odd length hex to be ok).   The change above makes the ishex*() set require a real hex string, prior to claiming that it is a hex string.   There was 1 format which then failed it's own ST code.  For that format, I added logic into the  valid which allowed success for   if (*sub_string == 0 || !ishex(sub_string))   goto fail